prow/plugins/trigger/pr.go

```diff
@@ -49,7 +49,26 @@ func handlePR(c client, trustedOrg string, pr github.PullRequestEvent) error {
 				return fmt.Errorf("could not welcome non-org member %q: %v", author, err)
 			}
 		}
-	case github.PullRequestActionReopened, github.PullRequestActionSynchronize:
+	case github.PullRequestActionReopened:
+		// When a PR is reopened, check that the user is in the org for now. If so, run the building.
+		orgMember, err := c.GitHubClient.IsMember(trustedOrg, author)
+		if err != nil {
+			return err
+		} else if orgMember {
+			comments, err := c.GitHubClient.ListIssueComments(pr.PullRequest.Base.Repo.Owner.Login, pr.PullRequest.Base.Repo.Name, pr.PullRequest.Number)
+			if err != nil {
+				return err
+			}
+			err = clearStaleComments(c.GitHubClient, trustedOrg, pr.PullRequest, comments)
+			if err != nil {
+				c.Logger.Warnf("Failed to clear stale comments: %v.", err)
+			}
+			// Just try to remove "needs-ok-to-test" label if existing, we don't care about the result.
+			c.GitHubClient.RemoveLabel(trustedOrg, pr.PullRequest.Base.Repo.Name, pr.PullRequest.Number, needsOkToTest)
+			c.Logger.Info("Starting all jobs for updated PR.")
+			return buildAll(c, pr.PullRequest)
```

Comment:
I'm not sure if it's strange to see the PR building when a non-new member reopen his/her PR.

Reply:
That is expected behavior.